test/nexus_test.go

```diff
@@ -31,6 +31,7 @@ import (
 	"testing"
 	"time"
 
+	"github.com/google/go-cmp/cmp"
```

Comment:
Is it worth a new test dependency just to show failures in this one test that should not happen anyways? I think https://pkg.go.dev/google.golang.org/protobuf/proto#Equal should be good enough for this case right?

Reply:
Removed